afanasy/src/watch/item.h

```diff
@@ -62,6 +64,9 @@ class Item
 	virtual void v_filesReceived( const af::MCTaskUp & i_taskup);
 
 	virtual bool calcHeight();
+	
+	/// Fills the provided menu with the available item's actions
+	virtual void generateMenu(QMenu & /*o_menu*/);
```

Comment:
`generateMenu()` looked to me like a quite general pattern (though I found some cases where the signature is different) so I put it all the way up to `Item` (grand-)parent class.